kin-core/src/main/java/kin/core/ServiceProvider.java

```diff
@@ -73,18 +63,25 @@ final public String getProviderUrl() {
         return providerUrl;
     }
 
-    final public int getNetworkId() {
-        return networkId;
+    final public String getNetworkId() {
+        return network.getNetworkPassphrase();
     }
 
     final public boolean isMainNet() {
-        return networkId == NETWORK_ID_MAIN;
+        return NETWORK_ID_MAIN.equals(network.getNetworkPassphrase());
     }
 
     final KinAsset getKinAsset() {
+        if (kinAsset == null) {
+            kinAsset = new KinAsset(getAssetCode(), getIssuerAccountId());
```

Comment:
Are we not afraid of multithreading?

Reply:
Good question, didn't take it into consideration. 
I think our approach should be that `KinClient`/`KinAccount` not thread-safe by default, as I don't see a point for that, if there will be a demand, so we should consider it in the future.